queue/src/main/java/io/kestra/queue/QueueFactory.java

```diff
@@ -0,0 +1,55 @@
+package io.kestra.queue;
+
+import io.kestra.core.utils.ExecutorsUtils;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.core.beans.BeanIntrospectionReference;
+import io.micronaut.core.io.service.ServiceDefinition;
+import io.micronaut.core.io.service.SoftServiceLoader;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+@Factory
+public class QueueFactory {
+    public final static String QUEUE_EXECUTOR = "queueExecutor";
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    public static <T extends GenericEvent> List<Class<T>> listAllEvent(ClassLoader classLoader, Class<T> eventClass) {
```

Comment:
Honestly, returning a hard-coded list of classes would have been much simpler and more explicit that using the bean introspection stuff of Micronaut.